# Test Logistic Regression:

#### 1. Use the breast-bin.csv dataset

In [1]:
from si.io.csv_file import read_csv

In [4]:
df_path = r"C:\Users\Mariana\si\datasets\breast-bin\breast-bin.csv"
df = read_csv(df_path, sep = ',', features = True, label = True) 

#### 2. Divide the dataset into train and test sets

In [5]:
from si.model_selection.split import train_test_split

In [6]:
train_dataset, test_dataset = train_test_split(df, test_size=0.33, random_state=42)

In [7]:
print(f'(As dimensões dos dados de treino correspondem a:', train_dataset.shape())
print(f'(As dimensões dos dados de teste correspondem a:',test_dataset.shape())

(As dimensões dos dados de treino correspondem a: (468, 9)
(As dimensões dos dados de teste correspondem a: (230, 9)


Os dados de treino apresentam 468 linhas e 9 colunas, enquanto que os dados de teste apresentam 230 linhas e 9 colunas.

#### 3. Train the model. Which score do you get? And the cost?

In [16]:
from si.models.logistic_regression import LogisticRegression

In [17]:
model = LogisticRegression()

In [18]:
model.fit(train_dataset)

In [19]:
print(model.score(test_dataset))

0.9826086956521739


O score de 0.9826 indica que o modelo obteve uma precisão de cerca de 98.26% no conjunto de dados de teste. Este valor sugere que o modelo utilizado é bastante bom para a classificação no conjunto de teste.

In [20]:
print(model.cost(test_dataset))

0.640935502202131
